src/System.Private.Xml/src/System/Xml/Xsl/XsltOld/DbgCompiler.cs

```diff
@@ -6,35 +6,29 @@ namespace System.Xml.Xsl.XsltOld
 {
     using System;
     using System.Collections;
-    using System.Xml;
+    using System.Diagnostics;
     using System.Xml.XPath;
     using System.Xml.Xsl.XsltOld.Debugger;
-    using System.Diagnostics;
-    using System.Runtime.Versioning;
 
     internal class DbgData
     {
-        private XPathNavigator _styleSheet;
         private VariableAction[] _variables;
-        public XPathNavigator StyleSheet { get { return _styleSheet; } }
-        public VariableAction[] Variables { get { return _variables; } }
+        public XPathNavigator StyleSheet { get; }
         public DbgData(Compiler compiler)
         {
             DbgCompiler dbgCompiler = (DbgCompiler)compiler;
-            _styleSheet = dbgCompiler.Input.Navigator.Clone();
+            StyleSheet = dbgCompiler.Input.Navigator.Clone();
             _variables = dbgCompiler.LocalVariables;
             dbgCompiler.Debugger.OnInstructionCompile(this.StyleSheet);
         }
         internal void ReplaceVariables(VariableAction[] vars) { _variables = vars; }
 
-        // static Empty:
-        private static DbgData s_nullDbgData = new DbgData();
         private DbgData()
         {
-            _styleSheet = null;
+            StyleSheet = null;
             _variables = Array.Empty<VariableAction>();
         }
-        public static DbgData Empty { get { return s_nullDbgData; } }
+        public static DbgData Empty { get; } = new DbgData();
```

Comment:
does this not need compiler param anymore?

Reply:
Not sure what you mean. The constructor with the compiler param is still there.

Reply:
@MisinformedDNA previously it was constructed like: `new DbgData(compiler)`

Reply:
Not sure when it was like that, but the history back to 2016 still shows it being called without parameters. See https://github.com/dotnet/corefx/blob/65cca50e20ad60f96527648db31ec03b9cc361aa/src/System.Private.Xml/src/System/Xml/Xsl/XsltOld/DbgCompiler.cs#L31-L37 If you feel this is still an issue, please file a ticket, but I didn't change the way `Empty` is constructed, so that shouldn't impact this PR, correct?

Reply:
@MisinformedDNA I'm looking at least one place in your PR: https://github.com/dotnet/corefx/pull/28989/files/774c6c5df9bf127ab998e31a42b9b962ee4ba60f#diff-e8fc15e8487936ea51029015f0919879L331
if you think it is redundant we can remove it, but I'd like to make sure we understand why we change that

Reply:
@MisinformedDNA what I'm saying is that in the call you removed you passed `compiler` argument and now you use parameterless constructor

Reply:
@krwq @huanwu

Reply:
cc: @pjanotti

Reply:
At first glance it seems to me that removing GetDbgData and related changes are partially changing the code related to DbgCompiler while keeping it. I don't see DbgCompiler being used, but it *seems*  useful if we ever need to debug anything in XSLT... 

At this point I would say to not touch it, if later we feel sure that we don't need this debugging code we should remove it all not only the specific method. That said I can be convinced otherwise...